src/OpenTelemetry.Exporter.Geneva/TLDExporter/TldTraceExporter.cs

```diff
@@ -247,61 +234,64 @@ internal void SerializeActivity(Activity activity)
         string statusDescription = string.Empty;
 
         int partCFieldsCountFromTags = 0;
-        var kvpArrayForPartCFields = PartCFields.Value;
-        if (kvpArrayForPartCFields == null)
-        {
-            kvpArrayForPartCFields = new KeyValuePair<string, object>[120];
-            PartCFields.Value = kvpArrayForPartCFields;
-        }
+        var kvpArrayForPartCFields = PartCFields.Value ??= new KeyValuePair<string, object>[120];
 
-        List<KeyValuePair<string, object>> envPropertiesList = null;
+        List<KeyValuePair<string, object?>>? envPropertiesList = null;
 
         foreach (ref readonly var entry in activity.EnumerateTagObjects())
         {
+            if (entry.Value == null)
+            {
+                continue;
+            }
+
             // TODO: check name collision
-            if (CS40_PART_B_MAPPING.TryGetValue(entry.Key, out string replacementKey))
+            if (CS40_PART_B_MAPPING.TryGetValue(entry.Key, out string? replacementKey))
             {
                 Serialize(eb, replacementKey, entry.Value);
                 partBFieldsCount++;
+                continue;
             }
-            else if (string.Equals(entry.Key, "otel.status_code", StringComparison.Ordinal))
+
+            if (entry.Key.StartsWith("otel.status_", StringComparison.Ordinal))
             {
-                if (string.Equals(Convert.ToString(entry.Value, CultureInfo.InvariantCulture), "ERROR", StringComparison.Ordinal))
+                var keyPart = entry.Key.AsSpan().Slice(12);
+                if (keyPart is "code")
```

Comment:
this is cool. I had to do some reading but this should be a allocation-free way of working with substrings. learned something new today :)

Reply:
Spans are magic 🤣 But this isn't so much about avoiding allocations as it is just speeding up the logic. If you look at the before it checked if the key was "otel.status_code" and then it checked if the key was "otel.status_description". What I did was switch it to see if the key starts with "otel.status_" if it doesn't, we know it isn't going to match and we can skip further processing. Only if the key starts with "otel.status_" then do we check for "code" or "description" essentially so we don't do any repeat work.

Reply:
nicely done!